libsolidity/analysis/OverrideChecker.cpp

```diff
@@ -587,7 +587,13 @@ void OverrideChecker::checkOverride(OverrideProxy const& _overriding, OverridePr
 		// This is only relevant for a function overriding a function.
```

Comment:
Actually this is a bug. We currently allow `external pure` functions to be overwritten by public state variables.

We should just assume that state mutability of a public state variable is `view` and then apply the same logic that we apply to functions here.

Reply:
I added the according logic & modified the tests. As this makes old code invalid, should we consider this a breaking change and I add the according commit in a different PR targeting `breaking`?